### PR TITLE
share_target dictionary in Manifest

### PR DESCRIPTION
**Pull request by cpyro**
Change references to the Manifest string field share_url_template to dictionary share_target with key url_template. Changes apply both to documentation, and demo Share Target Test App.

### CODE REVIEW

File: docs/interface.md [hunk 1]
```diff
@@ -36,21 +36,25 @@ shared data can be inserted into.
 
 ```WebIDL
 partial dictionary Manifest {
-  string share_url_template;
+  partial dictionary share_target {
```
Comment (mgiuca) [inline docs/interface.md]: You can't just put a dictionary in here AFAIK.

You need to define a separate dictionary (like if you were defining a struct in C) and then use it as a type name. Like we do with `ShareData` here:
https://github.com/WICG/web-share/blob/master/docs/interface.md

So make a "`dictionary ShareTarget`" and then here "`ShareTarget share_target`".

Also not "`partial dictionary`". "`partial`" refers to the fact that the rest of it is defined somewhere else, whereas you are defining your own full dictionary here.

**Comment by marcoscaceres** [comment]
@cpyro, unrelated, but the IRP bot is complaining that it can't work out your affiliation? Do you work for a w3c member? if not, that's totally ok... just trying to work out how to stop it annoying us.

**Comment by cpyro** [comment]
@marcoscaceres I've actually been trying to see what's going on; tried to log into w3.org this morning, and have reset my password twice, and waited a couple hours before trying to log in again, and it keeps saying my password is wrong :/
I am @mgiuca 's intern at Google, but this is my personal GitHub account. I also added Google to the GitHub company field in settings this morning, but to no effect.

**Comment by marcoscaceres** [comment]
Marked as non-substantive for IPR from ash-nazg.

**Comment by marcoscaceres** [comment]
Ah, ok. I'll mark these as non-substantive for now. And enjoy your internship! Matt and the team over there are awesome 👍

**Comment by cpyro** [comment]
Great! Thanks Marcos! Yeah, they are :D

File: docs/interface.md (hunk 1 above)
Comment (cpyro) [inline docs/interface.md]: Done.

**Pull request merged**